WordPress/src/main/java/org/wordpress/android/ui/reader/discover/interests/ReaderInterestsViewModel.kt

```diff
@@ -1,57 +1,151 @@
 package org.wordpress.android.ui.reader.discover.interests
 
+import androidx.annotation.StringRes
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import kotlinx.coroutines.launch
+import org.wordpress.android.R
+import org.wordpress.android.models.ReaderTag
 import org.wordpress.android.models.ReaderTagList
+import org.wordpress.android.ui.reader.discover.interests.ReaderInterestsViewModel.DoneButtonUiState.DoneButtonDisabledUiState
+import org.wordpress.android.ui.reader.discover.interests.ReaderInterestsViewModel.DoneButtonUiState.DoneButtonEnabledUiState
+import org.wordpress.android.ui.reader.discover.interests.ReaderInterestsViewModel.DoneButtonUiState.DoneButtonHiddenUiState
 import org.wordpress.android.ui.reader.repository.ReaderTagRepository
+import org.wordpress.android.viewmodel.Event
 import javax.inject.Inject
 
 class ReaderInterestsViewModel @Inject constructor(
     private val readerTagRepository: ReaderTagRepository
 ) : ViewModel() {
     var initialized: Boolean = false
 
-    private val _uiState = MutableLiveData<UiState>()
+    private val _uiState: MutableLiveData<UiState> = MutableLiveData(
+        UiState(listOf(), ReaderTagList(), DoneButtonHiddenUiState)
+    )
     val uiState: LiveData<UiState> = _uiState
 
+    private val _navigateToDiscover = MutableLiveData<Event<Unit>>()
+    val navigateToDiscover: LiveData<Event<Unit>> = _navigateToDiscover
+
+    private val selectedInterests = ReaderTagList()
+
     fun start() {
-        if (initialized) return
+        if (initialized) {
+            val uiState = uiState.value as UiState
+            updateUiState(uiState.copy(interestsUiState = getInterestsUiStateWithSelectedStates()))
```

Comment:
It is currently not possible to get list of selected ids from `ChipGroup` included with `material lib 1.1.0 ` https://git.io/JfX4j. So I'm saving selected items inside `selectedInterests`. Also `ChipGroup` doesn't seem to restore selected states for all chips on orientation change (just restores first selected chip), so I'm updating it with the saved selected states inside `onStart`. Let me know if you have any suggestion.

I can revisit this logic when we upgrade to `material lib ver >= v1.2.0-alpha01 ` where `ChipGroup` includes a [method to get the selected ids](https://git.io/JfXBq).

Reply:
I'm not sure I understand this part. Why do we need to update the uiState in `start`? AFAIU `start` is invoked after configuration change and `uiState: LiveData<UiState>` will retain its state, right?

Reply:
Refactored here: bfaaee9